### PR TITLE
Use rescript@v10

### PR DESCRIPTION
**Pull request by DZakh**

### CODE REVIEW

File: src/S.bs.js [hunk 1]
```diff
@@ -126,30 +126,30 @@ function toReason(nestedLevelOpt, error) {
...
-              return "- " + $$location + reason;
+              return "- " + $$location + "" + reason + "";
             });
         var reasons = Array.from(new Set(array));
-        return "Invalid union with following errors" + lineBreak + reasons.join(lineBreak);
+        return "Invalid union with following errors" + lineBreak + "" + reasons.join(lineBreak) + "";
     
   }
-  return "Expected " + reason.expected + ", received " + reason.received;
+  return "Expected " + reason.expected + ", received " + reason.received + "";
```
Comment (DZakh) [inline src/S.bs.js:144]: Started adding empty strings around values in template literal.

File: src/S.bs.js [hunk 2]
```diff
@@ -126,30 +126,30 @@ function toReason(nestedLevelOpt, error) {
...
 }
 
 function toString(error) {
   var match = error.operation;
   var operation = match ? "parsing" : "serializing";
   var reason = toReason(undefined, error);
   var pathText = formatPath(error.path);
-  return "[ReScript Struct]" + " Failed " + operation + " at " + pathText + ". Reason: " + reason;
+  return "[ReScript Struct] Failed " + operation + " at " + pathText + ". Reason: " + reason + "";
```
Comment (DZakh) [inline src/S.bs.js:152]: Here it joined to string values on compile-time, that's super cool!
(`[ReScript Struct]` and ` Failed `)

File: src/S.bs.js (hunk 1 above)
Comment (cristianoc) [reply src/S.bs.js:144]: Can't seem to repro:
```rescript
// tst.res
let template = (nestedLevel, reasons, foo) => {
  let lineBreak = `\n${" "->Js.String2.repeat(nestedLevel * 2)}`
  `Invalid union with following errors${lineBreak}${reasons->Js.Array2.joinWith(lineBreak)}`
}
```

gives

```js
// Generated by ReScript, PLEASE EDIT WITH CARE
'use strict';


function template(nestedLevel, reasons, foo) {
  var lineBreak = "\n" + " ".repeat((nestedLevel << 1));
  return "Invalid union with following errors" + lineBreak + reasons.join(lineBreak);
}

exports.template = template;
/* No side effect */
```

Would you open an issue with a self-contained repro?

Comment (DZakh) [reply src/S.bs.js:144]: I've created an issue https://github.com/rescript-lang/rescript-compiler/issues/5521

**Pull request closed without merging**